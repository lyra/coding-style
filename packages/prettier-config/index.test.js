const fs = require('fs')
const prettier = require('prettier')

describe('Validate Prettier configuration', () => {
  it('should format a file correctly', () => {
    return prettier.resolveConfig('index.json').then((options) => {
      const text = fs.readFileSync('__fixtures__/file-badly-formatted.tsx', 'utf8')
      const formatted = prettier.format(text, { ...options, parser: 'typescript' })
      expect(formatted).toMatchSnapshot()
    })
  })

  it('should organize imports correctly', () => {
    return prettier.resolveConfig('index.json').then((options) => {
      const text = fs.readFileSync('__fixtures__/file-disorganized-imports.ts', 'utf8')
      const formatted = prettier.format(text, { ...options, parser: 'typescript' })
      expect(formatted).toMatchSnapshot()
    })
  })
})
