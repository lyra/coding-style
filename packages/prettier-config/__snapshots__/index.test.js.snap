// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Validate Prettier configuration should format a file correctly 1`] = `
"// @ts-expect-error no webpack dep
// @ts-expect-error no react dep
import * as React from 'react'
import wonderful from 'wonderful'
// @ts-expect-error no relative dep
import { formatAge } from '../formats'

interface Props {
  name: string
  age: number
}

const Foo: React.FC<Props> = ({ name, age }) => {
  return (
    <div>
      <h1>My name is {wonderful(name)}.</h1>
      <h2>I am {formatAge(age)}.</h2>
    </div>
  )
}

export default Foo
"
`;
